File: backend/scripts/index_vectors.py
"""重建 sqlite-vec 知识索引（等价于原 TS scripts/index-vectors.ts）。

用法（在 backend/ 下）：python -m scripts.index_vectors
需要配置 AI_API_KEY 且 RAG_ENABLED=true。
"""
from __future__ import annotations

import asyncio

from app.config import settings
from app.data.ensure_vector_index import ensure_vector_index
from app.db import init_db


async def main() -> None:
    init_db()
    count = await ensure_vector_index(enabled=settings.RAG_ENABLED, api_key=settings.AI_API_KEY, force=True)
    print(f"indexed {count} passages")


if __name__ == "__main__":
    asyncio.run(main())

File: backend/scripts/index-vectors.ts
import { prisma } from '../src/lib/prisma'
import { env } from '../src/lib/env'
import { ensureVectorIndex } from '../src/data/ensureVectorIndex'

/**
 * 全量重建社团知识的向量索引（force）。
 * 用法：npm run index:vectors --workspace backend
 */
async function main() {
  if (!env.AI_API_KEY) {
    console.error('AI_API_KEY 未配置，无法生成向量索引。请在 backend/.env 中设置后重试。')
    process.exitCode = 1
    return
  }
  const count = await ensureVectorIndex(prisma, { enabled: true, apiKey: env.AI_API_KEY, force: true })
  console.log(`Indexed ${count} knowledge passages into sqlite-vec.`)
}

main()
  .catch(error => {
    console.error(error)
    process.exitCode = 1
  })
  .finally(() => prisma.$disconnect())
